rotkehlchen/inquirer.py

```diff
@@ -412,55 +430,117 @@ def _query_oracle_instances(
 
         if ignore_cache is False:
             cache = instance.get_cached_current_price_entry(cache_key=(from_asset, to_asset))
             if cache is not None:
-                return cache.price
+                return cache.price, cache.oracle
+
+        oracle_price, oracle_queried = instance._query_oracle_instances(from_asset=from_asset, to_asset=to_asset, skip_onchain=skip_onchain)  # noqa: E501
+        return oracle_price, oracle_queried
+
+    @staticmethod
+    def find_price(
+            from_asset: Asset,
+            to_asset: Asset,
+            ignore_cache: bool = False,
+            skip_onchain: bool = False,
+    ) -> Price:
+        """Wrapper around _find_price to ignore oracle queried when getting price"""
+        price, _ = Inquirer()._find_price(
+            from_asset,
+            to_asset,
+            ignore_cache,
+            skip_onchain,
+        )
+        return price
 
-        oracle_price = instance._query_oracle_instances(from_asset=from_asset, to_asset=to_asset, skip_onchain=skip_onchain)  # noqa: E501
-        return oracle_price
+    @staticmethod
+    def find_price_and_oracles(
+            from_asset: Asset,
+            to_asset: Asset,
+            ignore_cache: bool = False,
+            skip_onchain: bool = False,
+    ) -> Tuple[Price, Optional[CurrentPriceOracle]]:
+        """Wrapper around _find_price to include oracle queried when getting price"""
+        return Inquirer()._find_price(
+            from_asset,
+            to_asset,
+            ignore_cache,
+            skip_onchain,
+        )
 
     @staticmethod
     def find_usd_price(
             asset: Asset,
             ignore_cache: bool = False,
             skip_onchain: bool = False,
     ) -> Price:
+        """Wrapper around _find_usd_price to ignore oracle queried when getting usd price"""
+        price, _ = Inquirer()._find_usd_price(
+            asset,
+            ignore_cache,
+            skip_onchain,
+        )
+        return price
+
+    @staticmethod
+    def find_usd_price_and_oracles(
+            asset: Asset,
+            ignore_cache: bool = False,
+            skip_onchain: bool = False,
+    ) -> Tuple[Price, Optional[CurrentPriceOracle]]:
+        """Wrapper around _find_usd_price to include oracle queried when getting usd price"""
+        return Inquirer()._find_usd_price(
+            asset,
+            ignore_cache,
+            skip_onchain,
+        )
+
+    @staticmethod
+    def _find_usd_price(
+            asset: Asset,
+            ignore_cache: bool = False,
+            skip_onchain: bool = False,
+    ) -> Tuple[Price, Optional[CurrentPriceOracle]]:
         """Returns the current USD price of the asset
 
         Returns Price(ZERO) if all options have been exhausted and errors are logged in the logs
         """
         if asset == A_USD:
-            return Price(ONE)
+            return Price(ONE), CurrentPriceOracle.FIAT
```

Comment:
This shouldn't be `CurrentPriceOracle.FIAT` right? Given that it was not gotten from any oracle.